src/Microsoft.Azure.WebJobs.Host/Singleton/IDistributedLock.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Host
+{
+    /// <summary>
+    /// Handle for a lock returned by <see cref="IDistributedLockManager"/>
+    /// </summary>
+    public interface IDistributedLock
+    {
+        /// <summary>
+        /// The Lock identity.  
+        /// </summary>
+        string LockId { get; set; }
+
+        /// <summary>
+        /// Task is set to completed if the lease is lost (after it's been acquired). 
+        /// This allows consumers to get a notification if the lock failed to be renewed. 
+        /// </summary>
+        Task LeaseLost { get; }
```

Comment:
What is the benefit of using a Task here v.s. a regular event. For example a LockStatusChanged event with an event args of type DistributedLockStatusChangedEventArgs with a DistributedLock status enum, etc.